news/2016-10-10-tip2-devmode.asciidoc

```diff
@@ -0,0 +1,113 @@
+When enabled, a built-in tool called *Probe* provides a detailed view into internals of a CDI application.
+Probe makes it easy to inspect:
+
+* bean archives
+* beans and their properties such as qualifiers, stereotypes and name
+* a graph of bean dependencies
+* observers and producers declared by beans
+* interceptors and decorators bound to a bean
+* extensions
+* information about current contextual instances
+* tracking of invocations and invocation trees
+* tracking of fired events
+* Weld configuration
+* and more!
+
+Before you start further reading, please try out the link:http://probe-weld.itos.redhat.com/weld-numberguess/weld-probe[live demo hosted on OpenShift, window="_blank"] and feel the developer experience Weld is offering.
+Ok, now let's get it working!
+
+== How to enable the development mode
+
+No surprise that the development mode is _DISABLED_ by default.
+It should _NEVER_ be used in production as it may have negative impact on performance and/or represent a potential security risk.
+Make sure to disable the development mode before deploying to production!
+
+=== Web application
+
+For any web application, set the Servlet initialization parameter `org.jboss.weld.development` to `true`:
+
+[source,xml]
+----
+<web-app>
+    <context-param>
+        <param-name>org.jboss.weld.development</param-name>
+        <param-value>true</param-value>
+    </context-param>
+</web-app>
+----
+
+=== WildFly
+
+Since WildFly 10 it's also possible to enable the Weld development mode globally (i.e. for all the applications deployed) by setting `development-mode` attribute to true:
+----
+/subsystem=weld:write-attribute(name=development-mode,value=true)
+----
+See also link:https://docs.jboss.org/author/display/WFLY10/CDI+Reference[WildFly CDI Reference, window="_blank"].
+
+=== Weld SE
+
+For a Java SE application, set the system property `org.jboss.weld.development` to true:
+----
+java -cp myCoolApp.jar -Dorg.jboss.weld.development=true com.foo.MyMain
+----
+or use the `Weld.property()` method:
+
+[source,java]
+----
+org.jboss.weld.environment.se.Weld;
+
+public static void main(String[] args) {
+   Weld weld = new Weld().property("org.jboss.weld.development", true);
+   try (WeldContainer container = weld.initialize()) {
+      ...
+   }
+}
+----
+
+== Meet the Probe
+
+How does it actually work?
+Simply put: Probe collects some data about your application and then the data are available through the _REST API_, eventually through the _MXBean_ of name `org.jboss.weld.probe:type=JsonData,context=ID` where ID should be replaced with an idenfitier of an application.
+
+Probe has also a _web UI_ (single-page application) available at `{webappContextPath}/weld-probe` (once your webapp starts).
+Example URL: `http://localhost:8080/weld-numberguess/weld-probe`.
+We believe the UI is quite intuitive but feel free to ask questions on any channel (irc, gitter, mailing list, etc.).
+The best place to start is be probably the "Beans" view.
```

Comment:
I would try to mention more of the features here and describe them with one/two sentence(s) in order to present Probe's true powers. WDYT? I know it will get longer, but hey, Probe has a lot to offer, doesn't it?